api/pkg/backend/service.go

```diff
@@ -447,3 +460,342 @@ func (s *APIService) listStorageType(ctx context.Context, request *restful.Reque
+	result, err := s.backendClient.ListBackend(ctx, listBackendRequest)
+	if err != nil {
+		log.Errorf("failed to list backends: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	var failedBackends []string
+	for _, backendId := range tier.Backends {
+		exists := false
+		for _, backend := range result.Backends {
+			if backendId == backend.Id {
+				exists = true
+				break
+			}
+		}
+		if exists == false {
+			failedBackends = append(failedBackends, backendId)
+		}
+	}
+
+	if len(failedBackends) != 0 {
+		errMsg := fmt.Sprintf("invalid backends in request: %v are found", failedBackends)
+		log.Error(errMsg)
+		response.WriteError(http.StatusBadRequest, errors.New(errMsg))
+		return
+
+	}
+
+	res, err := s.backendClient.CreateTier(ctx, &backend.CreateTierRequest{Tier: tier})
+	if err != nil {
+		log.Errorf("failed to create tier: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	log.Info("Created backend successfully.")
+	response.WriteEntity(res.Tier)
+
+}
+
+//here backendId can be updated
+func (s *APIService) UpdateTier(request *restful.Request, response *restful.Response) {
+	log.Infof("Received request for updating tier: %v\n", request.PathParameter("id"))
+
+	id := request.PathParameter("id")
+	updateTier := backend.UpdateTier{Id: id}
+	body := ReadBody(request)
+	err := json.Unmarshal(body, &updateTier)
+	if err != nil {
+		log.Error("error occurred while decoding body", err)
+		response.WriteError(http.StatusBadRequest, nil)
+		return
+	}
+	if len(updateTier.AddBackends) == 0 && len(updateTier.DeleteBackends) == 0 {
+		log.Error("tier can not be updated with empty addBackends and deleteBackends")
+		response.WriteError(http.StatusBadRequest, err)
+		return
+	}
+
+	ctx := common.InitCtxWithAuthInfo(request)
+	res, err := s.backendClient.GetTier(ctx, &backend.GetTierRequest{Id: id})
+	if err != nil {
+		log.Errorf("failed to get tier details: %v\n", err)
+		err1 := errors.New("failed to get tier details")
+		response.WriteError(http.StatusInternalServerError, err1)
+		return
+	}
+
+	//duplicate  backends check:
+	visited := make(map[string]int, 0)
+	for _, val := range updateTier.AddBackends {
+		visited[val]++
+	}
+
+	for _, val := range updateTier.DeleteBackends {
+		visited[val]++
+	}
+
+	var duplicates []string
+	for u, v := range visited {
+		if v != 1 {
+			duplicates = append(duplicates, u)
+		}
+	}
+	if len(duplicates) != 0 {
+		errMsg := fmt.Sprintf("some backends in add or/and delete in request: %v are duplicate", duplicates)
+		log.Error(errMsg)
+		response.WriteError(http.StatusBadRequest, errors.New(errMsg))
+		return
+	}
+
+	//validation of add backends
+	listBackendRequest := &backend.ListBackendRequest{}
+	result, err := s.backendClient.ListBackend(ctx, listBackendRequest)
+	if err != nil {
+		log.Errorf("failed to list backends: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
```

Comment:
Everywhere in the code, please put more comments. Like at the start of for loop why this is being done

Reply:
added